eks-anywhere-common/Testers/newrelic/test-spec-configmap.yaml

```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-spec
+  namespace: newrelic
+data:
+  test-spec.yml: |-
+    description: |
+      End-to-end tests for k8s integration
+    custom_test_key: k8s.podName
+    scenarios:
+      - description: |
+          This scenario will verify that metrics from a k8s Cluster are correctly collected without privileges
+        tests:
```

Comment:
you may refer to https://github.com/xqi-nr/newrelic-integration-e2e-action#spec-file-for-the-e2e to verify more metric existence

Reply:
@anshrma Please see all the information you are looking for below. 
1. As you might observe all these pods that are `CrashLoopBackOff` or `ContainerCreating` is aligned to control plane nodes also pending pods are looking to schedule in control plane nodes. So your installation is looking to schedule pods in Control Plane which wont work because control plane exclusively, and we cannot use them to schedule pods. Please read this [AWS Blog](https://aws.amazon.com/blogs/aws/deploy-your-amazon-eks-clusters-locally-on-aws-outposts/#:~:text=Finally%2C%20I%20use,just%20like%20usual.) for more information. You should find ways to alter this through your helm release so the install dont schedule anything on control plane. 
2. To your question on test job, all test jobs are erroring, see the logs below.
3. Have answered the question 3 in point 1 above.

```bash
ubuntu@ip-10-0-0-178:~$ kubectl get nodes -o wide
NAME                                       STATUS     ROLES           AGE   VERSION                INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                KERNEL-VERSION                  CONTAINER-RUNTIME
ip-10-0-4-101.us-west-2.compute.internal   Ready      <none>          78d   v1.24.10-eks-48e63af   10.0.4.101    <none>        Amazon Linux 2                          5.10.167-147.601.amzn2.x86_64   containerd://1.6.6
ip-10-0-4-131.us-west-2.compute.internal   NotReady   control-plane   28d   v1.24.10-eks-08ad9cc   10.0.4.131    <none>        Bottlerocket OS 1.13.2 (aws-k8s-1.24)   5.15.90                         containerd://1.6.19+bottlerocket
ip-10-0-4-155.us-west-2.compute.internal   NotReady   control-plane   40d   v1.24.10-eks-08ad9cc   10.0.4.155    <none>        Bottlerocket OS 1.13.2 (aws-k8s-1.24)   5.15.90                         containerd://1.6.19+bottlerocket
ip-10-0-4-182.us-west-2.compute.internal   Ready      <none>          78d   v1.24.10-eks-48e63af   10.0.4.182    <none>        Amazon Linux 2                          5.10.167-147.601.amzn2.x86_64   containerd://1.6.6
ip-10-0-4-8.us-west-2.compute.internal     NotReady   control-plane   40d   v1.24.10-eks-08ad9cc   10.0.4.8      <none>        Bottlerocket OS 1.13.2 (aws-k8s-1.24)   5.15.90                         containerd://1.6.19+bottlerocket
ip-10-0-5-139.us-west-2.compute.internal   Ready      <none>          78d   v1.24.10-eks-48e63af   10.0.5.139    <none>        Amazon Linux 2                          5.10.167-147.601.amzn2.x86_64   containerd://1.6.6
ubuntu@ip-10-0-0-178:~$ kubectl get pods -n newrelic -o wide
NAME                                                        READY   STATUS              RESTARTS            AGE   IP           NODE                                       NOMINATED NODE   READINESS GATES
newrelic-newrelic-kube-state-metrics-78f4849688-7vvv9       1/1     Running             0                   42d   10.0.4.137   ip-10-0-4-101.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nri-kube-events-6bf76bddf7-84dgg          2/2     Running             0                   42d   10.0.4.112   ip-10-0-4-101.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nri-metadata-injection-6b7b47669d-vx7cf   0/1     Pending             0                   42d   <none>       <none>                                     <none>           <none>
newrelic-newrelic-nrk8s-controlplane-bllhr                  0/2     CrashLoopBackOff    18147 (47s ago)     40d   10.0.4.155   ip-10-0-4-155.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nrk8s-controlplane-f6flc                  1/2     CrashLoopBackOff    12401 (3m44s ago)   28d   10.0.4.131   ip-10-0-4-131.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nrk8s-controlplane-gj6rb                  0/2     CrashLoopBackOff    18155 (22s ago)     40d   10.0.4.8     ip-10-0-4-8.us-west-2.compute.internal     <none>           <none>
newrelic-newrelic-nrk8s-ksm-85d6bfc46b-wkrrk                0/2     ContainerCreating   0                   40d   <none>       ip-10-0-4-8.us-west-2.compute.internal     <none>           <none>
newrelic-newrelic-nrk8s-kubelet-26mgf                       0/2     ContainerCreating   0                   28d   <none>       ip-10-0-4-131.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nrk8s-kubelet-96nrm                       0/2     ContainerCreating   0                   40d   <none>       ip-10-0-4-155.us-west-2.compute.internal   <none>           <none>
newrelic-newrelic-nrk8s-kubelet-fnt4t                       0/2     Pending             0                   42d   <none>       <none>                                     <none>           <none>
newrelic-newrelic-nrk8s-kubelet-lz2r8                       0/2     Pending             0                   42d   <none>       <none>                                     <none>           <none>
newrelic-newrelic-nrk8s-kubelet-tbt4m                       0/2     Pending             0                   42d   <none>       <none>                                     <none>           <none>
newrelic-newrelic-nrk8s-kubelet-zl8r5                       0/2     ContainerCreating   0                   40d   <none>       ip-10-0-4-8.us-west-2.compute.internal     <none>           <none>
newrelic-testjob-28092130-2bxpw                             0/1     Error               0                   8h    10.0.4.53    ip-10-0-4-101.us-west-2.compute.internal   <none>           <none>
newrelic-testjob-28092130-zrwwb                             0/1     Error               0                   8h    10.0.4.4     ip-10-0-5-139.us-west-2.compute.internal   <none>           <none>
```

```bash
ubuntu@ip-10-0-0-178:~$ kubectl describe pod newrelic-newrelic-nri-metadata-injection-6b7b47669d-vx7cf -n newrelic 
Name:           newrelic-newrelic-nri-metadata-injection-6b7b47669d-vx7cf
----
Events:
  Type     Reason            Age                   From               Message
  ----     ------            ----                  ----               -------
  Warning  FailedScheduling  26m (x8189 over 27d)  default-scheduler  0/6 nodes are available: 3 Insufficient cpu, 3 node(s) had untolerated taint {node-role.eks-local.amazonaws.com/control-plane: }. preemption: 0/6 nodes are available: 3 No preemption victims found for incoming pod, 3 Preemption is not helpful for scheduling.
```

```bash
ubuntu@ip-10-0-0-178:~$ kubectl logs newrelic-newrelic-nrk8s-controlplane-bllhr -n newrelic 
Defaulted container "controlplane" out of: controlplane, forwarder
time="2023-05-31T18:27:18Z" level=info msg="Waiting for agent container to be ready..."
time="2023-05-31T18:28:48Z" level=error msg="creating integration wrapper: applying option: timeout waiting for agent: probe timed out after 1m30s"
```

```bash
ubuntu@ip-10-0-0-178:~$ k logs newrelic-newrelic-nrk8s-controlplane-f6flc -n newrelic 
Defaulted container "controlplane" out of: controlplane, forwarder
time="2023-05-31T18:28:32Z" level=info msg="Waiting for agent container to be ready..."
time="2023-05-31T18:30:02Z" level=error msg="creating integration wrapper: applying option: timeout waiting for agent: probe timed out after 1m30s"
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-ksm-85d6bfc46b-wkrrk  -n newrelic 
Events:
  Type     Reason           Age                        From     Message
  ----     ------           ----                       ----     -------
  Warning  NetworkNotReady  4m27s (x1770703 over 40d)  kubelet  network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-26mgf -n newrelic
Events:
  Type     Reason           Age                      From     Message
  ----     ------           ----                     ----     -------
  Warning  FailedMount      10m (x19829 over 28d)    kubelet  MountVolume.SetUp failed for volume "nri-integrations-cfg-volume" : object "newrelic"/"newrelic-newrelic-nrk8s-integrations-cfg" not registered
  Warning  NetworkNotReady  53s (x1209791 over 28d)  kubelet  network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-96nrm -n newrelic
Events:
  Type     Reason           Age                      From     Message
  ----     ------           ----                     ----     -------
  Warning  NetworkNotReady  58s (x1770540 over 40d)  kubelet  network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-fnt4t -n newrelic
Events:
  Type     Reason            Age                     From               Message
  ----     ------            ----                    ----               -------
  Warning  FailedScheduling  4m39s (x8196 over 27d)  default-scheduler  0/6 nodes are available: 1 Insufficient cpu. preemption: 0/6 nodes are available: 6 No preemption victims found for incoming pod.
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-lz2r8 -n newrelic 
Events:
  Type     Reason            Age                     From               Message
  ----     ------            ----                    ----               -------
  Warning  FailedScheduling  4m39s (x8196 over 27d)  default-scheduler  0/6 nodes are available: 1 Insufficient cpu. preemption: 0/6 nodes are available: 6 No preemption victims found for incoming pod.
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-tbt4m -n newrelic
Events:
  Type     Reason            Age                     From               Message
  ----     ------            ----                    ----               -------
  Warning  FailedScheduling  4m39s (x8196 over 27d)  default-scheduler  0/6 nodes are available: 1 Insufficient cpu. preemption: 0/6 nodes are available: 6 No preemption victims found for incoming pod.
```

```bash
ubuntu@ip-10-0-0-178:~$ k describe pod newrelic-newrelic-nrk8s-kubelet-zl8r5 -n newrelic
Events:
  Type     Reason           Age                        From     Message
  ----     ------           ----                       ----     -------
  Warning  NetworkNotReady  2m55s (x1771310 over 41d)  kubelet  network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized
```
```bash
ubuntu@ip-10-0-0-178:~$ k logs newrelic-testjob-28092130-zrwwb -n newrelic
Defaulted container "test-container" out of: test-container, kubectl (init)
Cloning into 'newrelic-integration-e2e-action'...
go: downloading github.com/sirupsen/logrus v1.9.0
go: downloading github.com/newrelic/newrelic-client-go v0.91.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading golang.org/x/sys v0.0.0-20220715151400-c0bba94af5f8
go: downloading github.com/google/go-querystring v1.1.0
go: downloading github.com/hashicorp/go-retryablehttp v0.7.0
go: downloading github.com/tomnomnom/linkheader v0.0.0-20180905144013-02ca5825eb80
go: downloading github.com/valyala/fastjson v1.6.3
go: downloading github.com/imdario/mergo v0.3.12
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/stretchr/testify v1.8.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
time="2023-05-31T10:21:59Z" level=info msg="running e2e"
time="2023-05-31T10:21:59Z" level=debug msg="parsing the content of the spec file"
time="2023-05-31T10:21:59Z" level=debug msg="return with settings"
time="2023-05-31T10:21:59Z" level=debug msg="validating the spec definition"
time="2023-05-31T10:21:59Z" level=debug msg="[scenario]: This scenario will verify that metrics from a k8s Cluster are correctly collected without privileges\n, [Tag]: newrelic-testjob-28092130-zrwwb"
time="2023-05-31T10:22:00Z" level=warning msg="Error detected" iteration=0
time="2023-05-31T10:22:00Z" level=error msg="querying: query did not return a valid result: SELECT latest(k8s.pod.startTime) FROM Metric SINCE 5 MINUTES AGO WHERE k8s.podName = 'newrelic-testjob-28092130-zrwwb'"
time="2023-05-31T10:22:05Z" level=fatal msg="after 1 attempts, last errors: [querying: query did not return a valid result: SELECT latest(k8s.pod.startTime) FROM Metric SINCE 5 MINUTES AGO WHERE k8s.podName = 'newrelic-testjob-28092130-zrwwb']"
exit status 1
```